metrics-operator/controllers/common/providers/elk/elk_test.go

```diff
@@ -0,0 +1,232 @@
+package elasticsearch
+
+import (
+	"context"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/elastic/go-elasticsearch/v8"
+	"github.com/go-logr/logr"
+	metricsapi "github.com/keptn/lifecycle-toolkit/metrics-operator/api/v1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+)
+
+func TestNewElasticProvider(t *testing.T) {
+	log := logr.Discard()
+	scheme := runtime.NewScheme()
+	k8sClient := fake.NewClientBuilder().WithScheme(scheme).Build()
+	elasticURL := "http://localhost:9200"
+
+	provider, err := NewElasticProvider(log, k8sClient, elasticURL)
+
+	assert.NoError(t, err)
+	assert.NotNil(t, provider)
+	assert.Equal(t, log, provider.Log)
+	assert.Equal(t, k8sClient, provider.K8sClient)
+	assert.NotNil(t, provider.Elastic)
+}
+
+func TestKeptnElasticProvider_FetchAnalysisValue(t *testing.T) {
+	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusOK)
+		response := `{
+			"hits": {
+				"total": {
+					"value": 42
+				}
+			}
+		}`
+		_, _ = w.Write([]byte(response))
+	}))
+	defer server.Close()
+
```

Comment:
let's try to mock the elastic to make the testing easier for multiple testing scenarios